### PR TITLE
Added example of extracting data from email in JSON format

### PR DESCRIPTION
**Pull request by bkakadiya**
# Description

The primary objective of this example is to transform raw email content into an easily interpretable JSON format, exemplifying a practical application of language models in data extraction

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/examples/usecases/email_data_extraction.ipynb
Comment (logan-markewich) [inline docs/examples/usecases/email_data_extraction.ipynb:1]: Is there anything the langchain loader is doing that the unstructured loader from llama-hub isn't doing?

```
from llama_index import download_loader

UnstructuredReader = download_loader('UnstructuredReader')
documents = UnstructuredReader().load_data("../data/email/ark-trading-jan-12-2024.msg")
content = documents[0].text
```

Comment (bkakadiya) [reply docs/examples/usecases/email_data_extraction.ipynb:1]: Thank your for your suggestion !!! I wasn't aware of it. 

I have updated to use llama-hub unstructured.

Thanks again !!

**Comment by hatianzhang** [comment]
could we make those 3 raw files downloadable thru google drive instead of put into repo

**Comment by bkakadiya** [comment]
> could we make those 3 raw files downloadable thru google drive instead of put into repo

@hatianzhang 
Each file is less than 80KB. Risk (minimal) that I see is, if someone accidently delete it from Google Drive,  notebook may not render and work properly. 

Is there anything specific advantage that you see to store it in google drive? Also point me to example that is pulling it from google drive or other public storage. 

Appreciate your feedback.

Thanks,
Bakul

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**